### ISSUE TITLE
ONNX support for AdaptiveMax/AvgPool ?

### ISSUE THREAD

**Issue by jaidmin** [pytorch/pytorch#5310]
Hi, 

I was just wondering whether it is possible/planned to add ONNX support for AdaptivePooling, because that enables using the same network with different resolutions (check https://github.com/fastai/fastai/blob/84d8ad253e1d3f5ecf1e1dcb07076e61f3f408b7/fastai/conv_learner.py#L40) for example.
When I wanted to export this it failed with 

`RuntimeError: ONNX export failed: Couldn't export Python operator AdaptiveMaxPool2d`

and I had to manually replace the AdaptivePoolings with normal ones, thus it now works with fixed sized inputs.

Cheers, Johannes

Edit: So in this case its possible to use GlobalPooling but there are still use cases for AdaptivePooling, where ONNX suppport would be very nice

**Comment by Scitator** [comment]
Hi @jaidmin  !
Is there any updates about this issue?
I have just the same problem - I use nn.AdaptiveMaxPool2d((1, 1)) and model becomes unconvertable. It would be great, if you post any hack to solve this problem.
Thanks!

**Comment by jaidmin** [comment]
Hi @Scitator !
Yes, as long as you are just using adaptive pooling to do global pooling you can simply define the following module:
```
class MyAdaptiveMaxPool2d(nn.Module):
    def __init__(self, sz=None):
        super().__init__()
       

    def forward(self, x): 
        inp_size = x.size()
        return nn.functional.max_pool2d(input=x,
                  kernel_size= (inp_size[2], inp_size[3]))
```

I also recently opened a discussion in the forums about how to replace adaptive pooling in general:
https://discuss.pytorch.org/t/how-to-replicate-f-adaptive-max-pool-using-f-max-pool/16851 

Cheers!

**Comment by Scitator** [comment]
@jaidmin Thanks, all works great! (even with pytorch 0.4)

**Comment by CoinCheung** [comment]
has this problem really been solved ? When I use tensorrt 7.0.0 to compile the onnx export from pytorch with nn.AvgPool2d, the I still met some problem if I use opset11, but the opset10 will be fine.
